src/cpp/common/alarm.cc

```diff
@@ -31,21 +31,17 @@
  */
 
 #include <grpc++/alarm.h>
-#include <grpc++/completion_queue.h>
 #include <grpc++/impl/grpc_library.h>
-#include <grpc/grpc.h>
 
 namespace grpc {
 
 static internal::GrpcLibraryInitializer g_gli_initializer;
-Alarm::Alarm(CompletionQueue* cq, gpr_timespec deadline, void* tag)
-    : tag_(tag),
-      alarm_(grpc_alarm_create(cq->cq(), deadline, static_cast<void*>(&tag_))) {
+
+Alarm::~Alarm() {
   g_gli_initializer.summon();
```

Comment:
Sorry, now that I read the code of summon, I have no problem. I thought it actually could do work but in this case was being masked by the fact that Alarm already has a CQ created. Now I see that it's a no-op to satisfy the linker, I have no problem.